File: Homework/Homework1VB/Homework1VB/Form1.vb
Imports System.Runtime.CompilerServices

Public Class Form1

    Dim count(5)

    Private Sub Button1_Click(sender As Object, e As EventArgs) Handles Button1.Click
        Dim max, min, a, s, d, f, g, h, arr(5)
        max = 6
        min = 1
        Randomize()
        For i = 0 To 5 Step 1
            arr(i) = Int((max - min + 1) * Rnd() + min)
        Next

        n1.Text = arr(0).ToString
        n2.Text = arr(1).ToString
        n3.Text = arr(2).ToString
        n4.Text = arr(3).ToString
        n5.Text = arr(4).ToString
        n6.Text = arr(5).ToString




        For i = 0 To 5 Step 1
            If arr(i) = 1 Then
                count(0) += 1
            ElseIf arr(i) = 2 Then
                count(1) += 1
            ElseIf arr(i) = 3 Then
                count(2) += 1
            ElseIf arr(i) = 4 Then
                count(3) += 1
            ElseIf arr(i) = 5 Then
                count(4) += 1
            ElseIf arr(i) = 6 Then
                count(5) += 1
            End If
        Next

        For Each series In Chart1.Series
            series.Points.Clear()
        Next



        Chart1.Series("Count").Points.AddXY("1", count(0))
        Chart1.Series("Count").Points.AddXY("2", count(1))
        Chart1.Series("Count").Points.AddXY("3", count(2))
        Chart1.Series("Count").Points.AddXY("4", count(3))
        Chart1.Series("Count").Points.AddXY("5", count(4))
        Chart1.Series("Count").Points.AddXY("6", count(5))











    End Sub

End Class

File: Homework/Homework5/Homework5/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Runtime.CompilerServices;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;
using static System.Security.Cryptography.RandomNumberGenerator;

namespace Homework5
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
            
        }

        Random r = new Random();
        public int Gen()
        {
            return r.Next(10, 289);
        }

        bool vert = true;

        private void button1_Click(object sender, EventArgs e)
        {
            if (vert)
            {
                this.chart1.Series["Dati"].ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Bar;
                vert = false;
            }
            else {
                this.chart1.Series["Dati"].ChartType = System.Windows.Forms.DataVisualization.Charting.SeriesChartType.Column;
                vert = true;
            }
            
        }

        private void button2_Click(object sender, EventArgs e)
        {
            int a = Gen();
            int s = Gen();
            this.pictureBox1.Width = a;
            this.pictureBox1.Height = s;
            foreach (var series in chart1.Series)
            {
                series.Points.Clear();
            }
            this.chart1.Series["Dati"].Points.AddXY("Base", a);
            this.chart1.Series["Dati"].Points.AddXY("Height", s);
            this.chart1.Series["Dati"].Points.AddXY("Diagonal", Math.Sqrt((a*a)+(s*s)));
            this.chart1.Series["Dati"].Points.AddXY("Perimeter", a+a+s+s);
            // this.chart1.Series["Dati"].Points.AddXY("Area", a*s);
        }
    }
}
